### DISCUSSION TITLE
Call to undefined function mb_strcut()

### DISCUSSION THREAD

**Discussion by bahiensed** [laravel/framework#40449] [Q&A] [answered]
- Laravel Version: 8.79.0
- PHP Version: 8.1.1
- Database Driver & Version: MariaDB

### Description:

Call to undefined function mb_strcut() (View: /var/www/application/vendor/laravel/framework/src/Illuminate/Mail/resources/views/html/layout.blade.php)

### Steps To Reproduce:

When creating a new user on route /register the error occurs. It creates the user in the database, but don't send the confirmation e-mail.

It works on Ubuntu 20.04.3 LTS (staging server), but not on Debian 11 (bullseye)(production server).
Both systems have a fresh install of OS, Laravel and Breeze.

live system on: http://app.bahien.se/register shows the error.

**Comment by crissi** [comment]
you need mbstring extension: 
https://www.php.net/manual/en/mbstring.installation.php
https://zoomadmin.com/HowToInstall/UbuntuPackage/php-mbstring

**Reply by Galihzen97** [reply]
how about using it on windows.?

**Comment by bahiensed** [comment]
> you need mbstring extension: https://www.php.net/manual/en/mbstring.installation.php https://zoomadmin.com/HowToInstall/UbuntuPackage/php-mbstring

it's already installed, as all other dependencies...

**Reply by Galihzen97** [reply]
how about using it on windows.?

**Comment by ahinkle** [comment]
Whats your output on:
```
php -m | grep mbstring
```

Honestly sounds extension-related and not Laravel - seems to be working fine with everything else 👍

**Reply by brayleB** [reply]
result is 
mbstring

**Comment by boutghatoua** [comment]
I'm having the same issue with the same parameters OS and Laravel/PHP versions and database;
the output of   `php -m | grep mbstring` is mbstring;

note that this occures when visiting routes that should send emails

**Comment by boutghatoua** [comment, marked as answer]
So after some inspections, it turns out that my server had two php versions installed 8.0 and 8.1
mbstring was installed on 8.1 while my laravel app was using php8.0

In my case to solve this I just removed all versions of php and only kept 8.1 and used it as default
( another problem I ran to was that pdo_mysql wasn't enabled so to enable it I used `phpenmod pdo_mysql` then restart apache2

Hope this helps

**Comment by rh-robin** [comment]
@dbahiense  
Did you solve it?
I am also facing this problem. If you solved then let me know how you did it.

**Reply by bahiensed** [reply]
Hi @rh-robin check again if mbstring is installed as other fellows said here. In my it was installed, but then I gave up and installed everything in Ubuntu instead of debian and in Ubuntu it worked out of the box. Greetings!

**Reply by rajeshtva** [reply]
@dbahiense also take care for installing correct php extension. for eg... if you are using `php 8.1` then install specifically `php8.1-mbstring php8.1-curl php8.1-zip` etc.. extension.. if you run `sudo apt-get install php-curl` then it is going to install generic operating system packaged version. which may not be the version you are currently running. learned it long way today. hope it helps..

**Comment by Galihzen97** [comment]
how about using it on windows?

**Reply by bigraja** [reply]
extension=mbstring enable

**Comment by Galihzen97** [comment]
Thank you verymuch

Pada tanggal Sen, 1 Mei 2023 11.31, bigraja ***@***.***>
menulis:

**Comment by yorunairu** [comment]
facing same problem on php 8.3.10 using aapanel on laravel 11
already installed php extension mbstring via terminal ubuntu or via app store aapanel still facing error
reload nginx, optimize laravel, still no good news.
any solution?
![image](https://github.com/user-attachments/assets/9d30913c-d7dc-4e0e-bcf8-6be4dc8506fa)

**Reply by henzeb** [reply]
first do `php -v` in a console to be sure you execute php 8.3, then follow with `php -m` and see if you can find `mb-string`.
